include/dlaf/util_matrix.h

```diff
@@ -21,149 +21,111 @@ constexpr double M_PI = 3.141592;
 #include <blas.hh>
 #include <hpx/hpx.hpp>
 
+#include "dlaf/common/assert.h"
 #include "dlaf/common/index2d.h"
 #include "dlaf/matrix.h"
 #include "dlaf/types.h"
 
 /// @file
 
 namespace dlaf {
-namespace util_matrix {
+namespace matrix {
+namespace util {
+namespace internal {
 
 /// @brief Verify if dlaf::Matrix is square
 ///
 /// @tparam Matrix refers to a dlaf::Matrix object
-/// @throws std::invalid_argument if the matrix is not squared
-template <class Matrix>
-void assertSizeSquare(const Matrix& matrix, std::string function, std::string mat_name) {
-  if (matrix.size().rows() != matrix.size().cols())
-    throw std::invalid_argument(function + ": " + "Matrix " + mat_name + " is not square.");
-}
+#define DLAF_PRECONDITION_SIZE_SQUARE(matrix)                                                         \
```

Comment:
Wouldn't `DLAF_ASSERT_SIZE_SQUARE` be a better name?
(and similar for next macros)